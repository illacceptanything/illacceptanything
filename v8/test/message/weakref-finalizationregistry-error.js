// Copyright 2020 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Flags: --expose-gc --noincremental-marking
// Flags: --no-stress-opt

// Since cleanup tasks are top-level tasks, errors thrown from them don't stop
// future cleanup tasks from running.

function callback(holdings) {
  throw new Error('callback');
};

const fg1 = new FinalizationRegistry(callback);
const fg2 = new FinalizationRegistry(callback);

(function() {
let x = {};
fg1.register(x, {});
fg2.register(x, {});
x = null;
})();

gc();
