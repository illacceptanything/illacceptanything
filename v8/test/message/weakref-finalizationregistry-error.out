*%(basename)s:{NUMBER}: Error: callback
  throw new Error('callback');
  ^
Error: callback
    at callback (*%(basename)s:{NUMBER}:{NUMBER})
    at FinalizationRegistry.cleanupSome (<anonymous>)

*%(basename)s:{NUMBER}: Error: callback
  throw new Error('callback');
  ^
Error: callback
    at callback (*%(basename)s:{NUMBER}:{NUMBER})
    at FinalizationRegistry.cleanupSome (<anonymous>)

